BlueprintUI/Sources/Element/ElementContent.swift

```diff
@@ -356,87 +387,119 @@ private struct EnvironmentAdaptingStorage: ContentStorage {
-        environment: Environment,
-        cache: CacheTree
-    ) -> [(identifier: ElementIdentifier, node: LayoutResultNode)] {
-        let environment = adapted(environment: environment)
-
-        let childAttributes = LayoutAttributes(size: attributes.bounds.size)
-
-        let identifier = ElementIdentifier(elementType: type(of: child), key: nil, count: 1)
-
-        let node = LayoutResultNode(
-            element: child,
-            layoutAttributes: childAttributes,
-            environment: environment,
-            children: child.content.performLayout(
-                attributes: childAttributes,
+        in size: CGSize,
+        with environment: Environment,
+        state: ElementState
+    ) -> [LayoutResultNode] {
+        state.layout(in: size, with: environment) { environment in
+            let environment = adapted(environment: environment)
+
+            let childAttributes = LayoutAttributes(size: size)
+
+            let identifier = ElementIdentifier.identifierFor(singleChild: child)
+
+            let childState = state.childState(for: child, in: environment, with: identifier)
+
+            let node = LayoutResultNode(
+                identifier: identifier,
+                layoutAttributes: childAttributes,
                 environment: environment,
-                cache: cache.subcache(element: child)
+                element: childState.element,
+                children: childState.elementContent.performLayout(
+                    in: size,
+                    with: environment,
+                    state: childState
+                )
             )
-        )
 
-        return [(identifier, node)]
+            return [node]
+        }
     }
 
-    func measure(in constraint: SizeConstraint, environment: Environment, cache: CacheTree) -> CGSize {
-        cache.get(constraint) { constraint -> CGSize in
-            let environment = adapted(environment: environment)
-            return child.content.measure(
+    func measure(
+        in constraint: SizeConstraint,
+        with environment: Environment,
+        state: ElementState
+    ) -> CGSize {
+        state.measure(in: constraint, with: environment) { environment in
+
+            let environment = self.adapted(environment: environment)
+            let identifier = ElementIdentifier.identifierFor(singleChild: child)
+            let childState = state.childState(for: child, in: environment, with: identifier)
+
+            return childState.elementContent.measure(
                 in: constraint,
-                environment: environment,
-                cache: cache.subcache(element: child)
+                with: environment,
+                state: childState
             )
         }
     }
 
     private func adapted(environment: Environment) -> Environment {
         var environment = environment
+
+        environment.readNotificationsEnabled = false
         adapter(&environment)
+        environment.readNotificationsEnabled = true
```

Comment:
Micronit (fine leaving as is): Should we add a function to `Environment` that allows you to customize it without sending read notifications so we can keep this property/behavior `private`?

Something like:
```
private func adapted(environment: Environment) -> Environment {
    environment.adaptedWithoutReadNotifications(adapter)
}
```

Reply:
I went back and forth on this vs the current shape, so happy to bring this one back

Reply:
We might also just kill this entirely; it's a pretty tiny micro-optimization.

Reply:
I'm just going to kill this for now.